### PR TITLE
Swap basic auth for token

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
HTTP Authentication

### Changes proposed in this pull request
Swap basic auth for token based auth

### Guidance to review
```
curl "api_endpoint_here"  -H "Authorization: Bearer bat"
```

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class ApplicationController < ActionController::API
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  before_action :authenticate
+
+  def authenticate
+    authenticate_or_request_with_http_token do |token|
+      AUTHENTICATION.include?(token)
```
Comment (misaka) [inline app/controllers/application_controller.rb]: In the docs for [ActionController::HttpAuthentication::Token](https://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token.html) they use `secure_compare` which sounds like worth doing to me.

Comment (misaka) [reply app/controllers/application_controller.rb]: Does it make sense to use `AUTHENTICATION` as a hash here? Also, you're checking with the keys, which are the usernames.

I think what we want here is a single secret, similar to what they have in the docs linked-to above.

Comment (dankmitchell) [reply app/controllers/application_controller.rb]: Ah yes, great find. Thanks @misaka

**Comment by dankmitchell** [comment]
[Token authorisation for the UCAS API](https://trello.com/c/3YyIOWlF/866-token-authorisation-for-the-ucas-api)

**Pull request merged**